### PR TITLE
adding  password change API endpoint

### PR DESCRIPTION
**Pull request by alex-kovoy**
**Purpose**

Added a password change endpoint for logged in users.

**Related Issues**

Fixes https://github.com/gravitational/teleport/issues/1390

### CODE REVIEW

**Review by klizhentas — APPROVED** [review]
this looks really nice

**Comment by alex-kovoy** [comment]
retest this please

File: lib/services/authority.go [hunk 1]
```diff
@@ -45,6 +46,20 @@ func (c *HostCertParams) Check() error {
 	return nil
 }
 
+// ChangePasswordReq defines a request to change user password
+type ChangePasswordReq struct {
```
Comment (russjones) [inline lib/services/authority.go]: Minor issue, this can probably live in `lib/auth/password.go`.

File: lib/web/password.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clt, err := c.GetClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	u2fReq, err := clt.GetU2FSignRequest(c.GetUser(), []byte(req.Pass))
```
Comment (russjones) [inline lib/web/password.go]: Use should be logged out here as well if they have been locked out. This endpoint essentially checks your password.

**Review by russjones — APPROVED** [review]
Overall looks great.

**Comment by alex-kovoy** [comment]
retest this please

**Comment by alex-kovoy** [comment]
closes #1390

**Pull request merged**